app/controllers/bookmarks_controller.rb

```diff
@@ -37,6 +37,8 @@ class BookmarksController < CatalogController
 
   self.add_show_tools_partial( :add_to_playlist, callback: :add_to_playlist_action )
 
+  self.add_show_tools_partial( :intercom_push, callback: :intercom_push_action, if: Proc.new { |context, config, options| context.user_can? :intercom_push } )
```

Comment:
Should this whole line have a conditional to avoid it loading unless Settings.intercom.present?

Reply:
I think because of timing of evaluation of this line, putting the conditional here didn't work. Reverting.